frontend/src/index.js

```diff
@@ -1,9 +1,38 @@
 import React from 'react'
 import ReactDOM from 'react-dom'
+import { setContext } from 'apollo-link-context'
+
+import {
+  ApolloClient,
+  ApolloProvider,
+  HttpLink,
+  InMemoryCache,
+} from '@apollo/client'
+
 import './index.css'
 import App from './App'
 
+const authLink = setContext((_, { headers }) => {
+  const token = localStorage.getItem('libraryapp-token')
+
+  return {
+    headers: {
+      ...headers,
+      authorization: token ? `bearer ${token}` : null,
+    },
+  }
+})
+
+const httpLink = new HttpLink({ uri: 'http://localhost:3001' })
+
+const client = new ApolloClient({
+  cache: new InMemoryCache(),
+  link: authLink.concat(httpLink),
+})
+
 ReactDOM.render(
-  <App />,
+  <ApolloProvider client={client}>
+    <App />
+  </ApolloProvider>,
```

Comment:
Wrap App as a child of ApolloProvider, so the client can be used in all components